### DISCUSSION TITLE
How to get a node location and

### DISCUSSION THREAD

**Discussion by krb-nantes** [idaholab/moose#26741] [Q&A General] [answered]
Hi all,

I am developing a code on MOOSE and what I need is to calculate the radial average at each axial division of part of my mesh, so I was thinking of getting the values of a variable on the inner wall node and on the outer wall node at the same axial position and take the average of both.

Primarily, the code where I will calculate this average looks like this:

```
  const Real inner_wall = 5
  std::vector nodelist = _mesh.getNodeList(inner_wall);
  for (int id : nodelist)
  {
      //get coordinate y of the node id

      //get node on outer wall at the same y
      out_node = Point(r_outer, y, 0.0) // not sure about this

      //get variable value at outer node
      out_var = _var[out_node]

      //get ch at inner node
      in_var = _var[id]

      // radial average
      const Real radial_avg = (in_var + out_var)/2
  }
```
So, I was wondering if it is possible to get a node axial location, knowing its ID, and the reverse, getting a node ID by its location.

**Comment by KhaledNabilSharafeldin** [comment, marked as answer]
Hi,
I have done something similar way back in an `action`, I'll share what I can remember here:
```
Point
getNodePos(libMesh::dof_id_type & n){
  auto nptr = *_mesh->nodePtr(n);
  return Point(nptr(0), nptr(1), nptr(2));
}
```
I used this to get xyz position of a known ID node, mind you the ID here is zero-based if I remember correctly.
So, If I wanted to get a bunch of coordinates from IDs:
```
std::vector<Point>
buildCoordVector(std::vector<libMesh::dof_id_type>& nodeID){
  std::vector<Point> out;
  for (auto &&i : nodeID)
    out.push_back(getNodePos(i));
  return out;
}
```
You can form a `nodeset` for the nodes of interest and that could be useful to what you're trying to do instead of looping over all the nodes in the domain to locate the nodes every time.

**Comment by GiudGiud** [comment]
Hello

I believe this is already implemented. 
You can use these two objects:
- parsedMaterial to turn the variable into a material property then
- a RadialAverage to compute the average
 https://mooseframework.inl.gov/source/userobjects/RadialAverage.html
 
 OR you can use this mesh division:
- CylindricalGridDivision to number the mesh https://mooseframework.inl.gov/source/meshdivisions/CylindricalGridDivision.html
- and this vector postprocessor to get a reduction using this numbering
https://mooseframework.inl.gov/source/vectorpostprocessors/MeshDivisionFunctorReductionVectorPostprocessor.html

**Reply by krb-nantes** [reply]
From those options, I think the first one may work better for me, however, when I tried to run the input with it I had the error:
```
A 'RadialAverage' is not a registered object.

If you are trying to find this object in a dynamically linked library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.
```

**Reply by GiudGiud** [reply]
I think this object is only present in a single module. Probably thermal_hydraulics.

maybe try the second option

**Reply by krb-nantes** [reply]
I just needed to update my MOOSE. Now, I can use the RadialAverage 😅
However, now I have two questions about it:
1- Does it calculate the same average for the whole block? (I need it for each axial position, in this case, it could be for each axial division of the mesh)
2- How can I retrieve the calculated value on the source code of my model and, if possible, on the input? For this last one, I looked on postprocessors and vectorpostprocessors and did not find any.
